shared-libs/rules-engine/src/transform-task-emission-to-doc.js

```diff
@@ -81,8 +81,8 @@ const minifyEmission = (taskDoc, emission) => {
   return taskDoc;
 };
 
-const newTaskDoc = (emission, userContactId, calculatedAt) => ({
-  _id: `task~${userContactId}~${emission._id}~${calculatedAt}`,
+const newTaskDoc = (emission, userSettingsId, calculatedAt) => ({
+  _id: `task~${userSettingsId}~${emission._id}~${calculatedAt}`,
```

Comment:
For the target doc we have both pieces of user data: `target~${docTag}~${userContactId}~${userSettingsId}` and here we have just the `userSettingsId` in the _id. I think it is fine as-is, but pointing out the inconsistency.